nemo/src/program_analysis/type_inference/position_graph.rs

```diff
@@ -158,16 +176,19 @@ impl PositionGraph {
                 {
                     for term in constructor.term().primitive_terms() {
                         if let PrimitiveTerm::Variable(body_variable) = term {
-                            let body_position = variables_to_last_node
-                                .get(body_variable)
-                                .expect("The iteration above went through all body atoms")
-                                .clone();
-
-                            graph.add_edge(
-                                body_position,
-                                head_position.clone(),
-                                PositionGraphEdge::BodyToHeadSameVariable,
-                            );
+                            // There is no entry in `variables_to_last_node` when the variable is an aggregate output variable, because in this case the variable is not in the body
+                            if let Some(body_position) = variables_to_last_node.get(body_variable) {
+                                graph.add_edge(
+                                    body_position.clone(),
+                                    head_position.clone(),
+                                    PositionGraphEdge::BodyToHeadSameVariable,
+                                );
+                            } else {
+                                debug_assert!(
+                                    matches!(body_variable, Variable::Universal(identifier) if identifier.0.starts_with(AGGREGATE_VARIABLE_PREFIX)),
+                                    "The iteration above went through all body atoms for non aggregate variables"
+                                )
+                            }
```

Comment:
While working on #435 i realized that there should be probably an egde being added here as well to connect the aggregate output variable to the constructor (output) variable. In this sense, the aggregate output variable behaves like a body variable I think.